### DISCUSSION TITLE
Can't access data from upstream data provider in nested data providers

### DISCUSSION THREAD

**Discussion by bastiaanterhorst** [Budibase/budibase#3938] [Help] [answered]
I have a structure like this:

```
- Repeater block with table set as data source
--- Container
----- Paragraph (displaying data from repeater block data source)
----- Data Provider with query set as data source (and property from repeater block as query parameter)
------- Paragraph (displaying data from data provider query
```

Unfortunately, it does not work. I simply get "an error has occured" without further info to debug.
When I change my query parameter from a dynamic property derived from the repeater block data source into a static number, it does work.

**Is it not possible to use properties from upstream data providers as downstream data provider parameters?**

If not, any idea how I would get solve a situation like this? Basically I am using a repeater to list items from a table, and then per item, I would like to show results based on a query I run for that specific item (so a query per row in the list).

Cheers!

**Comment by mjashanks** [comment, marked as answer]
HI Bastiaan. This should be possible. I think you are missing a `Repeater`, after your final `Data Provider`. (or you could use a `Repeater Block` instead of the final `Data  Provider`... so

```
- Repeater block with table set as data source
--- Container
----- Paragraph (displaying data from repeater block data source)
----- REPEATER BLOCK with query set as data source (and property from repeater block as query parameter)
------- Paragraph (displaying data from data provider query
```

**Comment by bastiaanterhorst** [comment]
It works! Thanks for the help.
